internal/provider/function_rfc3339_parse_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/knownvalue"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+func TestRFC3339Parse_UTC(t *testing.T) {
+	resource.UnitTest(t, resource.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			// TODO: Replace with the stable v1.8.0 release when available
+			tfversion.SkipBelow(version.Must(version.NewVersion("v1.8.0-alpha20240228"))),
```

Comment:
My plan prior to merging this PR, update this and all other alpha references to the newly released beta version, then merge with this TODO for a later follow-up PR 😄

Reply:
Oh interesting, does the version checking not work with the below during prereleases?

```
        TerraformVersionChecks: []tfversion.TerraformVersionCheck{
            tfversion.SkipBelow(tfversion.Version1_8_0),
        },
```

Reply:
Yeah, `tfversion.SkipBelow(tfversion.Version1_8_0)` skips the prereleases. It made sense initially to me because I was thinking of `v1.8.0-alpha20240228` as being below/before `v1.8.0`.

Is that not the intended behavior?

Reply:
I asked a question while looking up the answer lol, here is the handling of pre-release comparisons: https://github.com/hashicorp/go-version/blob/644291d14038339745c2d883a1a114488e30b702/version.go#L123-L138

Reply:
🙁 well that's unfortunate. Maybe can adjust those terraform-plugin-testing implementations to ignore prerelease information as I'm guessing its going to be a much more common use case to want to allow prereleases that match the rest of the version information. I'll raise an issue over there.